### PR TITLE
Expressions using Transformations

### PR DESCRIPTION
**Pull request by Nathan-Fenner**
Through the usage of `GetTransformation`, expression is now capable of performing transformation invocations.

### CODE REVIEW

**Comment by jeeyoungk** [comment]
LGTM

File: query/expression.go [hunk 1]
```diff
@@ -201,6 +201,35 @@ func (expr *functionExpression) Evaluate(context EvaluationContext) (value, erro
...
+		first, err := expr.arguments[0].Evaluate(context)
+		if err != nil {
+			return nil, err
+		}
+		list, err := first.toSeriesList(context.Timerange)
+		if err != nil {
+			return nil, err
+		}
+		// Evaluate all the other parameters:
+		rest := expr.arguments[1:]
```
Comment (achow) [inline query/expression.go]: Isn't this dead code since we enforce `len(expr.arguments) == 1`?

Comment (Nathan-Fenner) [inline query/expression.go]: Aggregates have the requirement that there is *only* one argument; transformations require *at least* one argument. Individual transformations each check how many more are expected. For example, moving average has a second one for the time over which to average, while derivative wants only the series.

File: query/expression.go [hunk 2]
```diff
@@ -201,6 +201,35 @@ func (expr *functionExpression) Evaluate(context EvaluationContext) (value, erro
 		return seriesListValue(series), nil
 	}
 
+	if transform, ok := GetTransformation(name); ok {
+		//Verify that at least one argument is given.
+		if len(expr.arguments) != 1 {
```
Comment (achow) [inline query/expression.go]: Looks like you need to fix this then.

Comment (Nathan-Fenner) [inline query/expression.go]: Ah, yes

File: query/transformation.go [hunk 3]
```diff
@@ -193,3 +193,24 @@ func transformMapMaker(name string, fun func(float64) float64) func([]float64, [
...
+}
+
+func GetTransformation(name string) (transform, bool) {
+	transform, ok := transformTable[name]
+	return transform, ok
+}
+
+func NewTransformation(name string, transform transform) {
+	if _, ok := transformTable[name]; ok {
+		panic(fmt.Sprintf("transformation `%s` has already been declared", name))
```
Comment (achow) [inline query/transformation.go]: We shouldn't be using panics explicitly... This is easily a recoverable error.

panics should be reserved for catastrophic exceptions.

**Pull request merged**